### PR TITLE
Update documentation about contributing

### PR DESCRIPTION
**Pull request by travisbrown**
Problem

The contributing documentation currently recommends submitting pull requests
against master.

Solution

Update the contributing documentation to reflect the current workflow, and add
a note about branches to the main README.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -7,30 +7,34 @@ We'd love to get patches from you!
...
-3.  Make your cool new feature or bugfix on your branch
-4.  Write a test for your change
-5.  From your branch, make a pull request against twitter/finagle/master
-6.  Work with repo maintainers to get merged in
-7.  Wait for your change to be pulled into twitter/finagle/master
-8.  Merge twitter/finagle/master into your origin master
-9.  Delete your feature branch
+2.  Check out the `develop` branch
+3.  Make a feature branch
+4.  Make your cool new feature or bugfix on your branch
```
Comment (kevinoliver) [inline CONTRIBUTING.md]: should we mention what you need to do in order to build off of util as well?

Comment (travisbrown) [inline CONTRIBUTING.md]: I guess we probably should, although once the new round of releases comes out it shouldn't be necessary for a while, and once we've started publishing snapshots it'll be unnecessary for good. I could go either way.

Comment (kevinoliver) [inline CONTRIBUTING.md]: ok, lets leave it as is.

**Comment by kevinoliver** [comment]
lgtm

**Comment by stevegury** [comment]
LGTM

**Comment by luciferous** [comment]
looks good!

**Comment by travisbrown** [comment]
[Merged](https://github.com/twitter/finagle/commit/4cbd42db29197692191855630904bb3a5ea98021).

**Pull request closed without merging**